cl/settings/project/security.py

```diff
@@ -73,3 +75,60 @@
     "OPTIONS",
 )
 CORS_ALLOW_CREDENTIALS = True
+
+# CSP
+# Components:
+# - hCaptcha: https://docs.hcaptcha.com/#content-security-policy-settings
+# - Plausible: https://github.com/plausible/docs/issues/20
+# - Stripe: https://stripe.com/docs/security/guide#content-security-policy
+CSP_CONNECT_SRC = (
+    "'self'",
+    "https://hcaptcha.com/",
+    "https://*.hcaptcha.com",
+    "https://plausible.io/",
+    "https://api.stripe.com",
+)
+CSP_FRAME_SRC = (
+    "'self'",
+    "https://hcaptcha.com/",
+    "https://*.hcaptcha.com",
+    "https://js.stripe.com",
+    "https://hooks.stripe.com",
+)
+CSP_IMG_SRC = (
+    "'self'",
+    f"https://{AWS_S3_CUSTOM_DOMAIN}/",
+    "data:",  # @tailwindcss/forms uses data URIs for images.
+)
+CSP_OBJECT_SRC = (
+    "'self'",
+    f"https://{AWS_S3_CUSTOM_DOMAIN}/",
+)
+CSP_SCRIPT_SRC = (
+    "'self'",
+    "'report-sample'",
+    f"https://{AWS_S3_CUSTOM_DOMAIN}/",
+    "https://hcaptcha.com/",
+    "https://*.hcaptcha.com",
+    "https://plausible.io/",
+    "https://js.stripe.com",
+)
+CSP_STYLE_SRC = (
+    "'self'",
+    "'report-sample'",
+    f"https://{AWS_S3_CUSTOM_DOMAIN}/",
+    "https://hcaptcha.com/",
+    "https://*.hcaptcha.com",
+    "'unsafe-inline'",
+)
+CSP_DEFAULT_SRC = ("'self'",)
+CSP_BASE_URI = "'none'"
+CSP_INCLUDE_NONCE_IN = ["script-src"]
+if not any(
+    (DEVELOPMENT, TESTING)
+):  # Development and test aren’t used over HTTPS (yet)
+    CSP_UPGRADE_INSECURE_REQUESTS = True
+if SENTRY_REPORT_URI:
+    CSP_REPORT_URI = SENTRY_REPORT_URI
+CSP_REPORT_ONLY = True
+CSP_EXCLUDE_URL_PREFIXES = "/api"
```

Comment:
Well, one thing that's interesting about DRF is that by default it returns JSON, but if you use the `content-type` header, it respects that. Therefore, if you browse to our API, you get HTML and if you use curl without any headers, you get the default (JSON). 

I think this means having CSP on the API makes sense so that the HTML responses are safe. 

What's the argument for excluding it from the API? (Sorry to be a bit naive here.)

Reply:
Only argument against including it is mostly aesthetic – too many belt and suspenders. But CSP is all about defense in depth. There’s _no_ harm in applying it universally. I’ll patch shortly.